frontend/packages/dev-console/src/utils/pipeline-utils.ts

```diff
@@ -307,3 +307,21 @@ export const getPipelineRunParams = (pipelineParams: PipelineParam[]): PipelineR
     }))
   );
 };
+
+export const pipelineRunDuration = (run: PipelineRun): string => {
+  const startTime = _.get(run, ['status', 'startTime'], null);
+  const completionTime = _.get(run, ['status', 'completionTime'], null);
+
+  // Duration cannot be computed if start time is missing or a completed/failed pipeline has no end time
+  if (!startTime || (!completionTime && pipelineRunStatus(run) !== 'Running')) {
+    return '-';
+  }
+  const start = new Date(startTime).getTime();
+
+  // For running pipelines duration must be current time - starttime
+  const duration = completionTime
+    ? new Date(completionTime).getTime() - start
+    : new Date().getTime() - start;
```

Comment:
This can result in incorrect values due to different time zones.

Reply:
Missed that. Will update

Reply:
Let me know. This was just a guess.

Reply:
updated this, yes for the current time the browser will pick the local time indeed. Thanks for pointing this out